index.d.ts

```diff
@@ -1,2 +1,2 @@
-declare function isEqual<A = any, B = any>(a: A, b: B): boolean;
+declare function isEqual<B = any>(a: any, b: B): a is B;
```

Comment:
Great idea, converting this to a typeguard!  Solid improvement.

It doesn't seem like there's any possible way to overload this function to return both `a is B & b is A`, since the typeguard syntax is very limiting.  

So the big question, as Grant said, is whether we should return `a is B` versus `b is A`.  This is purely a semantical choice, since the actual order of the params does not matter.

IMO, the current implementation, `a is B`, seems like the right choice.  If I was describing the logic in my own words, I'd say something like "if a is equal to b, then do stuff with a".  Basically, the first argument is the "subject", so it makes sense that I'd intend to interact with the subject next.

Reply:
@gksander  Should we consider this Type-only change a breaking change? Major, minor, or patch?

Reply:
That's a really good question 🤔 On the one hand, a change to TS types could theoretically break a `typecheck` process, but doesn't affect runtime behavior. My gut reaction is a minor release, but my opinions on that aren't terribly strong. What do you think?

Reply:
I think semver applies strongly to runtime behavior, but for static type checks, I prefer patch.  This can only change compile-time behavior, which is extremely easy to catch and fix, so it doesn't warrant any extra warnings.

Also, this typeguard change will result in **wider** types, because it implicitly widens the type of a variable, so typically that means no changes will be necessary.

The only thing I can think of, where this change could also require code changes, is for scenarios where you've already "widened" the scope manually (via a cast), and perhaps you have checks like [no-unnecessary-type-assertion](https://typescript-eslint.io/rules/no-unnecessary-type-assertion) which will then complain about the unnecessary cast.  This is actually ideal, since it will force you to remove the redundancy.

Long story short, this change should be really easy to adopt, so let's go with `patch` :)

Reply:
Love it. @dwilt will you rebase with (or merge in) `master` branch and then run `yarn changeset` and add a `patch` changeset to this PR? That way once we get this merged, we can release ASAP.

Reply:
Done!